### PR TITLE
Mobile class pages part 1

### PR DESCRIPTION
**Pull request by puneetjohal**
# Purpose

###### A brief description of the purpose of the PR's changes for the project.
The first part of the mobile class page epic.

<br>

# Tickets

###### A link to any tickets this PR is associated with on Trello.\

- https://trello.com/c/BibeeNlf/295-mobile-class-pages-part-1

# Contributors

###### Anyone who contributed to this PR for future reference.

- @puneetjohal 
- @andrewydai 

# Feature List

###### A list of more in-depth and technical changes, additions, deletions, and fixes this PR provides.

image
image

# Notes

- The expandables with more info (i.e. recent professors, recent semesters offered) are currently empty
- The notification button doesn't do anything

###### A list of miscellaneous notes for this pull request, such as whether we need to add something later for production, or that it was time boxed, etc.

<br>

# Checklist

- [x] Filled out PR template :wink:
- [x] Approved by designers
- [x] Is passing linting checks
- [x] Is passing tsc
- [x] Is passing existing tests
- [x] Has documentation and comments in code
- [x] Has test coverage for code
- [x] Will have clear squash commit message

<br>
@sandboxnu/searchneu

### CODE REVIEW

File: components/ClassPage/mobile/MobileClassInfoExpandable.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import { GetClassPageInfoQuery } from '../../../generated/graphql';
+import IconCollapseExpand from '../../icons/IconCollapseExpand';
+
+export type MobileClassInfoExpandableProps = {
+  classPageInfo: GetClassPageInfoQuery;
+  title: string;
+};
+
+export default function MobileClassInfoExpandable({
+  classPageInfo,
```
Comment (hankewyczz) [inline components/ClassPage/mobile/MobileClassInfoExpandable.tsx:11]: This argument doesn't seem to be used?

File: components/ClassPage/mobile/MobileClassInfoExpandable.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  const [expanded, setExpanded] = useState(false);
+
+  return (
+    <div className="mobileClassInfoExpandable">
+      <div
+        className={
+          expanded
+            ? 'mobileClassInfoExpandable__header--expanded'
+            : 'mobileClassInfoExpandable__header'
+        }
```
Comment (hankewyczz) [inline components/ClassPage/mobile/MobileClassInfoExpandable.tsx]:
```suggestion
        className={"mobileClassInfoExpandable__header" + (expanded ? "--expanded" : "")}
```

File: components/ClassPage/mobile/MobilePageContent.tsx [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+              >
+                <Markup content={classPageInfo.class.latestOccurrence.desc} />
+              </div>
+              <div
+                className="classPageContainer__panel--showMore"
+                role="button"
+                tabIndex={0}
+                onClick={() => setShowMore(!showMore)}
+              >
+                {showMore ? 'Show less' : 'Show more'}
```
Comment (hankewyczz) [inline components/ClassPage/mobile/MobilePageContent.tsx]: This seems a little confusingly named - (if `showMore` is true, doesn't it seem like 'Show more' should be returned?). Maybe `showingMore` would be more explicit

File: components/ClassPage/mobile/MobilePageContent.tsx [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+                  {`Course Fees: ${
+                    classPageInfo.class.latestOccurrence.feeAmount
+                      ? `$${classPageInfo.class.latestOccurrence.feeAmount.toLocaleString()}`
+                      : 'None'
+                  }`}
+                </span>
+              </div>
+
+              <div className="avgInfo">
+                <p>In this course, there is, on average:</p>
```
Comment (hankewyczz) [inline components/ClassPage/mobile/MobilePageContent.tsx]:
```suggestion
                <p>In this course, there are, on average:</p>
```

File: components/ClassPage/mobile/MobilePageContent.tsx [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+                  </li>
+                  <li>
+                    <b>{Math.round(mean(seatsFilled(classPageInfo)))}</b> seats
+                    filled, or{' '}
+                    <b>
+                      {(
+                        Math.round(mean(seatsFilled(classPageInfo))) /
+                        Math.round(mean(seatsAvailable(classPageInfo)))
+                      ).toFixed(1)}
+                      %
```
Comment (hankewyczz) [inline components/ClassPage/mobile/MobilePageContent.tsx:149]: If you want this to be a percent, multiply by 100 - right now it's just a decimal 

example:
image

**Review by hankewyczz — CHANGES_REQUESTED** [review]
Looks good! 

Besides the comments I left, I have one issue - the grad/undergrad classification.

Take CS5800 (grad Algo) - it shows as [undergraduate](https://searchneu-95yuw09wj-sandboxneu.vercel.app/NEU/202230/classPage/CS/5800). However, on Banner, the course says that "Must be enrolled in one of the following Classifications: Graduate"

We don't have a good way of knowing this is a grad course from our API, so the best thing might just be to change the phrasing (instead of saying "Course Level", maybe just list the college?)

File: components/ClassPage/mobile/MobileClassInfoExpandable.tsx (hunk 1 above)
Comment (puneetjohal) [reply components/ClassPage/mobile/MobileClassInfoExpandable.tsx:11]: it will be in the next iteration!

**Comment by puneetjohal** [comment]
> We don't have a good way of knowing this is a grad course from our API, so the best thing might just be to change the phrasing (instead of saying "Course Level", maybe just list the college?)

The course level thing is a vestige of the desktop class pages, but I agree with this feedback. @andrewydai thoughts on changing course level to college on both the mobile and desktop class pages?

**Review by hankewyczz — APPROVED** [review]
🚀  looks good to me

**Pull request merged**